### PR TITLE
HIPRTC phaseout - Update color_convert and tensor_transpose

### PR DESCRIPTION
**Pull request by r-abishek**

### CODE REVIEW

**Review by kiritigowda — APPROVED** [review]

**Comment by r-abishek** [comment]
> @paveltc can you please run unit tests on this.
> @r-abishek please remove all the warnings

@rrawther The codacy warnings are removed

File: src/modules/hip/hip_computer_vision.cpp [hunk 1]
```diff
@@ -905,6 +905,148 @@ harris_corner_detector_hip_batch(Rpp8u *srcPtr, Rpp8u *dstPtr,rpp::Handle& handl
...
+    in_strides[3] = 1;
+
+    out_strides[0] = out_dims[1] * out_dims[2] * out_dims[3];
+    out_strides[1] = out_dims[2] * out_dims[3];
+    out_strides[2] = out_dims[3];
+    out_strides[3] = 1;
+
+    Rpp32u *d_perm, *d_in_strides, *d_out_strides, *d_out_dims;
+    hipMalloc(&d_perm, 4 * sizeof(Rpp32u));
+    hipMalloc(&d_in_strides, 4 * sizeof(Rpp32u));
```
Comment (rrawther) [inline src/modules/hip/hip_computer_vision.cpp:967]: Why do you need all the strides in hipMemory?

File: src/modules/hip/kernel/tensor.cpp [hunk 2]
```diff
@@ -366,6 +366,102 @@ RppStatus hip_exec_tensor_matrix_multiply(Rpp8u *srcPtr1, Rpp8u *srcPtr2, Rpp8u
...
+                       d_perm,
+                       d_out_strides,
+                       d_in_strides);
+
+    return RPP_SUCCESS;
+}
+
+RppStatus hip_exec_tensor_transpose_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, Rpp32u *d_out_dims, Rpp32u *d_perm, Rpp32u *d_out_strides, Rpp32u *d_in_strides, Rpp32u *out_dims, rpp::Handle& handle)
+{
+    // int localThreads_x = 16;
```
Comment (rrawther) [inline src/modules/hip/kernel/tensor.cpp:395]: Why is this commented?

**Review by rrawther — CHANGES_REQUESTED** [review]
@r-abishek : please resolve review comments

File: src/modules/hip/hip_computer_vision.cpp (hunk 1 above)
Comment (r-abishek) [reply src/modules/hip/hip_computer_vision.cpp:967]: The stride for each dimension in the 4D tensor was put in an array that needs to be accessed inside the kernel with the respective [0/1/2/3] index.

File: src/modules/hip/kernel/tensor.cpp (hunk 2 above)
Comment (r-abishek) [reply src/modules/hip/kernel/tensor.cpp:395]: FP16 for HIP isn't resolved yet

**Comment by paveltc** [comment]
This PR passed the unit tests.

**Review by rrawther — APPROVED** [review]

**Pull request merged**